### PR TITLE
projects/Amlogic: update.sh fix and cleanup

### PR DESCRIPTION
**Pull request by kszaq**
If there is no file matching `/flash/*.dtb` wildcard, the loop will proceed with `all_dtb="/flash/*.dtb"`, causing erratic behaviour in next script steps. Fix this by checking if  exists before proceeding.

### CODE REVIEW

File: projects/Amlogic/bootloader/update.sh [hunk 1]
```diff
@@ -75,12 +75,14 @@ for arg in $(cat /proc/cmdline); do
...
-        dtb=$(basename $all_dtb)
-        if [ -f $SYSTEM_ROOT/usr/share/bootloader/$dtb ]; then
-          echo "*** updating Device Tree Blob: $dtb ..."
-          mount -o rw,remount $BOOT_ROOT
-          cp -p $SYSTEM_ROOT/usr/share/bootloader/$dtb $BOOT_ROOT
+      for all_dtb in /flash/*.dtb ; do
+        if [ -f $all_dtb ] ; then
+          dtb=$(basename $all_dtb)
+          if [ -f $SYSTEM_ROOT/usr/share/bootloader/$dtb ]; then
+            echo "*** updating Device Tree Blob: $dtb ..."
```
Comment (chewitt) [inline projects/Amlogic/bootloader/update.sh]: Please use sentence casing so `Updating device tree blob:` and avoid `***` prefixes with on-screen messages as we don't includes them with the kernel and system messages. There are a couple of other `***` instances that need to be removed too.

Comment (kszaq) [reply projects/Amlogic/bootloader/update.sh]: The patch is simply a bugfix, the only added thing is `if`: https://github.com/LibreELEC/LibreELEC.tv/pull/2502/files?w=1
"*** updating Device Tree Blob: $dtb ..." follows what was present in official LE 8.2: 
https://github.com/LibreELEC/LibreELEC.tv/blob/libreelec-8.2/packages/tools/u-boot/scripts/update-c2.sh#L38-L45
https://github.com/LibreELEC/LibreELEC.tv/blob/libreelec-8.2/packages/tools/u-boot/scripts/update.sh#L42-L50
If this requires a cleanup, I suggest it's done in a separate PR.

**Comment by kszaq** [comment]
The PR is not is any way critical,  I will add changes requested by @chewitt to it later today.

**Comment by MilhouseVH** [comment]
I know this is just a bug fix so something else to consider for the future PR is the number of times that `mount -o rw,remount $BOOT_ROOT` is performed - I see it being called in 5 places, and some of those are within a `for` loop... would it not be better to mount `rw` once (unconditionally) at the beginning, and then at the end of the script restore `ro`?

**Comment by kszaq** [comment]
PR updated with updated messages in `update.sh` and `mount -o rw,remount` cleanup.

**Pull request merged**